src/statusBarTerminal.ts

```diff
@@ -2,24 +2,41 @@ import * as vscode from 'vscode';
 
 export class StatusBarTerminal {
     private _item: vscode.StatusBarItem;
-    private _terminal: vscode.Terminal;
+    private _showing: Boolean = false;
+    public _terminal: vscode.Terminal;
 
     constructor(terminalIndex: number, name?: string) {
         this._item = vscode.window.createStatusBarItem();
-        this.setTerminalIndex(terminalIndex);
+        this.setTerminalIndex(terminalIndex, name);
         this._item.show();
 
         this._terminal = vscode.window.createTerminal(name);
-        this._terminal.show();
+        this.show();
+    }
+
+    get name() {
+        return this._terminal.name;
     }
 
     public show() {
+        this._showing = true;
+        this._item.color = "yellow";
```

Comment:
Maybe this color should be a setting? Some users may not like the very bright yellow.